test-utils/src/dev/clickhouse.rs

```diff
@@ -141,6 +106,153 @@ impl ClickHouseInstance {
         })
     }
 
+    /// Start a new replicated ClickHouse server on the given IPv6 port.
+    pub async fn new_replicated(
+        port: String,
+        tcp_port: String,
+        interserver_port: String,
+        name: String,
+        r_number: String,
+        config_path: PathBuf,
+    ) -> Result<Self, anyhow::Error> {
+        let data_dir = TempDir::new()
+            .context("failed to create tempdir for ClickHouse data")?;
+        let log_path = data_dir.path().join("clickhouse-server.log");
+        let err_log_path = data_dir.path().join("clickhouse-server.errlog");
+        let tmp_path = data_dir.path().join("tmp/");
+        let user_files_path = data_dir.path().join("user_files/");
+        let access_path = data_dir.path().join("access/");
+        let format_schemas_path = data_dir.path().join("format_schemas/");
+        let args = vec![
+            "server".to_string(),
+            "--config-file".to_string(),
+            format!("{}", config_path.display()),
+        ];
+
+        let child = tokio::process::Command::new("clickhouse")
+            .args(&args)
+            .stdin(Stdio::null())
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .env("CLICKHOUSE_WATCHDOG_ENABLE", "0")
+            .env("CH_LOG", &log_path)
+            .env("CH_ERROR_LOG", err_log_path)
+            .env("CH_REPLICA_DISPLAY_NAME", name)
+            .env("CH_LISTEN_ADDR", "::")
+            .env("CH_LISTEN_PORT", port.clone())
+            .env("CH_TCP_PORT", tcp_port)
+            .env("CH_INTERSERVER_PORT", interserver_port)
+            .env("CH_DATASTORE", data_dir.path())
+            .env("CH_TMP_PATH", tmp_path)
+            .env("CH_USER_FILES_PATH", user_files_path)
+            .env("CH_USER_LOCAL_DIR", access_path)
+            .env("CH_FORMAT_SCHEMA_PATH", format_schemas_path)
+            .env("CH_REPLICA_NUMBER", r_number)
+            // There seems to be a bug using ipv6 with a replicated set up
+            // when installing all servers and coordinator nodes on the same
+            // server. For this reason we will be using ipv4 for testing.
```

Comment:
What's the nature of this bug? Will it affect us if we happen to deploy two replicas on the same host?

Reply:
It only appears when using localhost in the same server. It shouldn't be a problem for any rack installation.